### ISSUE TITLE
gloo net add `.json` convenience method for `Request`

### ISSUE THREAD

**Issue by databasedav** [rustwasm/gloo#199]
## Summary

add `.json` convenience method for `Request`

## Motivation

makes sending json parameters more ergonomic

## Detailed Explanation

right now, we must do a `.body(serde_json::to_string(&JsonSerializableStruct { ... }).unwrap())`, which isn't as ergonomic for an http library as, for example, [`reqwest`](https://docs.rs/reqwest/latest/reqwest/struct.RequestBuilder.html#method.json) with which you can do `.json(&JsonSerializableStruct { ... })`

## Drawbacks, Rationale, and Alternatives

i don't think there are drawbacks, refer to [`reqwest source`](https://docs.rs/reqwest/latest/src/reqwest/async_impl/request.rs.html#436) for implementation

## Unresolved Questions

**Comment by databasedav** [comment]
you also have to add an `.header("Content-Type", "application/json")` right now, which would be embedded in the method

**Comment by hamza1311** [comment]
Would you like to PR an implementation?

**Comment by minauteur** [comment]
If something like the following would suffice (not sure given that the surrounding methods return `Self` rather than `Result`), I can start adding some tests and open a PR:

```
    /// convenience method for json request bodies
    pub fn json<T: Serialize + ?Sized>(mut self, value: &T) -> Result<Self,Error> {
        let mut error = None;

            match serde_json::to_string(value) {
                Ok(s) => {
                    self = self.header("Content-Type", "application/json");
                    self = self.body(&s);
                }
                Err(err) => error = Some(Error::SerdeError(err)),
            };
        if let Some(err) = error {
            return Result::Err(err);
        }
        Result::Ok(self)
    }
```

**Comment by minauteur** [comment]
This can probably be closed, as the requested change was added in #204
